go.mod

```diff
@@ -0,0 +1,53 @@
+module github.com/jetstack/kube-oidc-proxy
+
+go 1.12
+
+require (
+	github.com/Masterminds/semver v1.4.2
+	github.com/golang/mock v0.0.0-20160127222235-bd3c8e81be01
+	github.com/heptiolabs/healthcheck v0.0.0-20180807145615-6ff867650f40
+	github.com/sirupsen/logrus v1.4.1
+	github.com/spf13/cobra v0.0.5
+	gopkg.in/square/go-jose.v2 v2.3.1
+	k8s.io/api v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/apiserver v0.0.0
+	k8s.io/cli-runtime v0.0.0
+	k8s.io/client-go v11.0.0+incompatible
+	k8s.io/component-base v0.0.0
+	k8s.io/klog v0.3.3
+	k8s.io/kubernetes v0.0.0-00010101000000-000000000000
+	sigs.k8s.io/kind v0.0.0-00010101000000-000000000000
+)
+
+replace (
+	k8s.io/api => k8s.io/api v0.0.0-20190620084959-7cf5895f2711
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20190620085554-14e95df34f1f
+	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190612205821-1799e75a0719
+	k8s.io/apiserver => k8s.io/apiserver v0.0.0-20190620085212-47dc9a115b18
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.0.0-20190620085706-2090e6d8f84c
+	k8s.io/client-go => k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20190219215739-d1e3091efae0
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20190620090013-c9a0fc045dc1
+	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20190311093542-50b561225d70
+	k8s.io/component-base => k8s.io/component-base v0.0.0-20190620085130-185d68e6e6ea
+	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20190531030430-6117653b35f1
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20190620090116-299a7b270edc
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20190620085325-f29e2b4a4f84
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20190620085942-b7f18460b210
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20190228160746-b3a7cee44a3
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20190620085809-589f994ddf7f
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20190620085912-4acac5405ec6
+	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20190602132728-7075c07e78bf
+	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20190620085838-f1cb295a73c9
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190620090156-2138f2c9de18
+	k8s.io/metrics => k8s.io/metrics v0.0.0-20190620085625-3b22d835f165
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190620085408-1aef9010884e
+	sigs.k8s.io/structured-merge-diff => sigs.k8s.io/structured-merge-diff v0.0.0-20190302045857-e85c7b244fd2
```

Comment:
Looks like you are pinning a lot more than is needed here - you can remove all of these that are not listed in the `require` block above 😄